src/Coin.cpp

```diff
@@ -6,221 +6,155 @@
 
 #include "Coin.h"
 
-#include "Aeternity/Address.h"
-#include "Aion/Address.h"
-#include "Algorand/Address.h"
-#include "Binance/Address.h"
-#include "Bitcoin/Address.h"
-#include "Bitcoin/CashAddress.h"
-#include "Bitcoin/SegwitAddress.h"
-#include "Cardano/AddressV3.h"
-#include "Cosmos/Address.h"
-#include "Decred/Address.h"
-#include "EOS/Address.h"
-#include "Ethereum/Address.h"
-#include "FIO/Address.h"
-#include "Filecoin/Address.h"
-#include "Groestlcoin/Address.h"
-#include "Harmony/Address.h"
-#include "Icon/Address.h"
-#include "IoTeX/Address.h"
-#include "Kusama/Address.h"
-#include "NEAR/Address.h"
-#include "NULS/Address.h"
-#include "Nano/Address.h"
-#include "Nebulas/Address.h"
-#include "Nimiq/Address.h"
-#include "Ontology/Address.h"
-#include "Polkadot/Address.h"
-#include "Ripple/Address.h"
-#include "Ripple/XAddress.h"
-#include "Solana/Address.h"
-#include "Stellar/Address.h"
-#include "TON/Address.h"
-#include "Tezos/Address.h"
-#include "Tron/Address.h"
-#include "Wanchain/Address.h"
-#include "Waves/Address.h"
-#include "Zcash/TAddress.h"
-#include "Zilliqa/Address.h"
-#include "NEO/Address.h"
-
+#include "CoinEntry.h"
 #include <TrustWalletCore/TWHRP.h>
 
-#pragma clang diagnostic push
-#pragma clang diagnostic fatal "-Wswitch"
+#include <map>
+#include <set>
+
+// Includes for entry points for coin implementations
+#include "Aeternity/Entry.h"
+#include "Aion/Entry.h"
+#include "Algorand/Entry.h"
+#include "Bitcoin/Entry.h"
+#include "Binance/Entry.h"
+#include "Cardano/Entry.h"
+#include "Cosmos/Entry.h"
+#include "Decred/Entry.h"
+#include "EOS/Entry.h"
+#include "Ethereum/Entry.h"
+#include "Filecoin/Entry.h"
+#include "FIO/Entry.h"
+#include "Groestlcoin/Entry.h"
+#include "Harmony/Entry.h"
+#include "Icon/Entry.h"
+#include "IoTeX/Entry.h"
+#include "Kusama/Entry.h"
+#include "Nano/Entry.h"
+#include "NEAR/Entry.h"
+#include "Nebulas/Entry.h"
+#include "NEO/Entry.h"
+#include "Nimiq/Entry.h"
+#include "NULS/Entry.h"
+#include "Ontology/Entry.h"
+#include "Polkadot/Entry.h"
+#include "Ripple/Entry.h"
+#include "Solana/Entry.h"
+#include "Stellar/Entry.h"
+#include "Tezos/Entry.h"
+#include "Theta/Entry.h"
+#include "TON/Entry.h"
+#include "Tron/Entry.h"
+#include "VeChain/Entry.h"
+#include "Wanchain/Entry.h"
+#include "Waves/Entry.h"
+#include "Zcash/Entry.h"
+#include "Zilliqa/Entry.h"
```

Comment:
I wonder if we can insert `#include "NewCoin/Entry.h"` and setup dispatcher automatically when we run `newcoin` script? same like we did for `TWCoinType.h`